package/Dockerfile

```diff
@@ -7,16 +7,21 @@ RUN mkdir -p /image/etc/ssl/certs /image/run /image/var/run /image/tmp /image/li
 
 FROM scratch
 ARG VERSION="dev"
+ARG PROG="k3s"
+
 COPY --from=base /image /
 RUN mkdir -p /etc && \
     echo 'hosts: files dns' > /etc/nsswitch.conf && \
-    echo "PRETTY_NAME=\"K3s ${VERSION}\"" > /etc/os-release && \
+    echo "PRETTY_NAME=\"${PROG} ${VERSION}\"" > /etc/os-release && \
     chmod 1777 /tmp
 VOLUME /var/lib/kubelet
-VOLUME /var/lib/rancher/k3s
+VOLUME /var/lib/rancher/${PROG}
 VOLUME /var/lib/cni
 VOLUME /var/log
 ENV PATH="$PATH:/bin/aux"
-ENV CRI_CONFIG_FILE="/var/lib/rancher/k3s/agent/etc/crictl.yaml"
-ENTRYPOINT ["/bin/k3s"]
+ENV CRI_CONFIG_FILE="/var/lib/rancher/${PROG}/agent/etc/crictl.yaml"
+ENV PROG=${PROG}
+ADD ./package/entrypoint.sh /bin
```

Comment:
Can we add the entrypoint to the `base` layer up above? Adding content here creates an additional layer on the shipping image.